ci/impalamgr.py

```diff
@@ -244,50 +244,61 @@ def load(data, udf, data_dir, overwrite):
 
     # load the data files
     if data:
-        tmp_dir = tempfile.mkdtemp(prefix='__ibis_tmp_')
-        try:
-            load_impala_data(con, str(data_dir), overwrite)
-        finally:
-            rm('-rf', tmp_dir)
+        load_impala_data(con, str(data_dir), overwrite)
     else:
-        print('Skipping Ibis test data load (--no-data)')
+        logger.info('Skipping Ibis test data load (--no-data)')
 
     # build and upload the UDFs
     if udf:
         already_loaded = is_udf_loaded(con)
-        print('Attempting to build and load test UDFs')
+        logger.info('Attempting to build and load test UDFs')
         if already_loaded and not overwrite:
-            print('UDFs already loaded and not overwriting; moving on')
+            logger.info('UDFs already loaded and not overwriting; moving on')
         else:
             if already_loaded:
-                print('UDFs already loaded; attempting to overwrite')
-            print('Building UDFs')
+                logger.info('UDFs already loaded; attempting to overwrite')
+            logger.info('Building UDFs')
             build_udfs()
-            print('Uploading UDFs')
+            logger.info('Uploading UDFs')
             upload_udfs(con)
     else:
-        print('Skipping UDF build/load (--no-udf)')
+        logger.info('Skipping UDF build/load (--no-udf)')
 
 
 def load_impala_data(con, data_dir, overwrite=False):
     already_loaded = is_impala_loaded(con)
-    print('Attempting to load Ibis Impala test data (--data)')
+    logger.info('Attempting to load Ibis Impala test data (--data)')
     if already_loaded and not overwrite:
-        print('Data is already loaded and not overwriting; moving on')
+        logger.info('Data is already loaded and not overwriting; moving on')
     else:
         if already_loaded:
-            print('Data is already loaded; attempting to overwrite')
+            logger.info('Data is already loaded; attempting to overwrite')
 
-        print('Uploading to HDFS')
+        logger.info('Uploading to HDFS')
         upload_ibis_test_data_to_hdfs(con, data_dir)
-        print('Creating Ibis test data database')
+
+        logger.info('Creating Ibis test data database')
         create_test_database(con)
-        parquet_tables = create_parquet_tables(con)
-        avro_tables = create_avro_tables(con)
-        for table in parquet_tables + avro_tables:
-            print('Computing stats for', table.op().name)
+
+        def compute_stats(table):
+            logger.info('Computing stats for %s', table.op().name)
             table.compute_stats()
 
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            parquet_tables = create_parquet_tables(con, executor)
+            avro_tables = create_avro_tables(con, executor)
+            list(
```

Comment:
It's a bit nested, decompose it into two expressions with a temporary variable?

Reply:
Sure